### DISCUSSION TITLE
How do I style the taglist_squares?

### DISCUSSION THREAD

**Discussion by Akshaybun** [awesomeWM/awesome#3828] [Q&A] [answered]
It seems that `taglist_squares_sel`  requires an image.
So I asked chatgpt for the solution and it gave me the following code for image generation.
It works but the result seems to be low res, no matter the imagesurface res. 
```lua
local function create_taglist_square(color_border, color_fill)
    local surface = cairo.ImageSurface(cairo.Format.ARGB32, 10, 10)
    local cr = cairo.Context(surface)

    -- Calculate the dimensions for a 4x4 square
    local size = 4

    cr:set_line_width(1) -- Adjust the line width as needed

    -- Fill the inside with the specified color_fill
    cr:set_source(gears.color(color_fill))
    cr:rectangle(3, 0, size, size) -- Adjust the rectangle position and size as needed
    cr:fill()

    -- Draw the border with the specified color_border
    cr:set_source(gears.color(color_border))
    cr:rectangle(3, 0, size, size) -- Adjust the rectangle position and size as needed
    cr:stroke()

    return surface
end
```
I want it to be just like the default one, I only need to change the border color and filled color of these squares. Can anyone help me how do I do that?

**Comment by Akshaybun** [comment, marked as answer]
Ok so stupid enough I found this
 ```lua
local taglist_square_size = dpi(4)
theme.taglist_squares_sel = theme_assets.taglist_squares_sel(
    taglist_square_size, theme.fg_normal
)
```
Also, Can I change the placement of this square?

**Reply by actionless** [reply]
> Also, Can I change the placement of this square?

that would require using custom widget template, as in the example at the top of this page: https://awesomewm.org/apidoc/widgets/awful.widget.taglist.html
